pysr/sr.py

```diff
@@ -2586,3 +2583,190 @@ def _mutate_parameter(param_name: str, param_value):
         return False
 
     return param_value
+
+
+class PySRSequenceRegressor(PySRRegressor):
+    def __init__(
+        self,
+        recursive_history_length: int = 0,
+        **kwargs,
+    ):
+        self.recursive_history_length = recursive_history_length
+        super().__init__(**kwargs)
+
+    def fit(
+        self,
+        X,
+        Xresampled=None,
+        weights=None,
+        variable_names: Optional[ArrayLike[str]] = None,
+        complexity_of_variables: Optional[
+            Union[int, float, List[Union[int, float]]]
+        ] = None,
+        X_units: Optional[ArrayLike[str]] = None,
+    ) -> "PySRRegressor":
+        """
+        Search for equations to fit the time series dataset and store them in `self.equations_`.
+
+        Parameters
+        ----------
+        X : ndarray | pandas.DataFrame
+            Training time series data of shape (n_samples, 1) or (1, n_samples).
+        Xresampled : ndarray | pandas.DataFrame
+            Resampled training data, of shape (n_resampled, 1) or (1, n_resampled),
+            to generate a denoised data on. This
+            will be used as the training data, rather than `X`.
```

Comment:
I don't think this functionality is compatible with the sequence regressor so it should just be removed. (If it is compatible, then it should have a test. But I feel like it's probably not compatible.)